### PR TITLE
pkg/tinycrypt: Fix dependencies

### PR DESCRIPTION
**Pull request by maribu**
### Contribution description

`tinycrypt` according to the test application only runs on 32 bit archs. Rather than using a whitelist in the test as ugly hack, simply model the requirements using `FEATURES_REQUIRED += arch_32bit` in the package.

### Testing procedure

`make -C tests/pkg_tinycrypt info-boards-supported` should only yield 32 bit architectures.

### Issues/PRs references

Stumbled upon while reviewing https://github.com/RIOT-OS/RIOT/pull/15241

### CODE REVIEW

File: pkg/tinycrypt/Makefile.dep [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Cifra is only supported by 32 bit architectures
```
Comment (bergzand) [inline pkg/tinycrypt/Makefile.dep:1]: Copy/past mistake?

```suggestion
# TinyCrypt is only supported by 32 bit architectures
```

Comment (maribu) [reply pkg/tinycrypt/Makefile.dep]: Yep. But let's just drop the comment. The line below is pretty self-explaining

**Review by bergzand — APPROVED** [review]
Ack

**Comment by maribu** [comment]
Backport provided in #15244

**Pull request merged**